src/test/java/io/cdap/plugin/servicenow/sink/RecordToJsonTransformerTest.java

```diff
@@ -0,0 +1,105 @@
+
+  @Test
+  public void testTransformWithValue() {
+    Schema schema = Schema.recordOf("record",
+                                    Schema.Field.of("id", Schema.of(Schema.Type.LONG)),
+                                    Schema.Field.of("price", Schema.of(Schema.Type.DOUBLE)),
+                                    Schema.Field.of("stockSize", Schema.of(Schema.Type.INT)),
+                                    Schema.Field.of("updated", Schema.of(Schema.Type.BOOLEAN)),
+                                    Schema.Field.of("batchId", Schema.of(Schema.Type.STRING)),
+                                    Schema.Field.of("tax", Schema.of(Schema.Type.FLOAT)),
+                                    Schema.Field.of("timestampMillis",
+                                                    Schema.nullableOf(Schema.of(Schema.LogicalType.TIMESTAMP_MILLIS))),
+                                    Schema.Field.of("timeMillis",
+                                                    Schema.nullableOf(Schema.of(Schema.LogicalType.TIME_MILLIS))),
+                                    Schema.Field.of("timestamp",
+                                                    Schema.nullableOf(Schema.of(Schema.LogicalType.TIMESTAMP_MICROS))),
+                                    Schema.Field.of("timeMicros",
+                                                    Schema.nullableOf(Schema.of(Schema.LogicalType.TIME_MICROS))),
+                                    Schema.Field.of("date",
+                                                    Schema.nullableOf(Schema.of(Schema.LogicalType.DATE))),
+                                    Schema.Field.of("dateTime",
+                                                    Schema.nullableOf(Schema.of(Schema.LogicalType.DATETIME))),
+                                    Schema.Field.of("updatedPrice", (Schema.decimalOf(5, 2))));
+
+    StructuredRecord record = Mockito.mock(StructuredRecord.class);
+    Mockito.when(record.get("date")).thenReturn("20220610");
+    Mockito.when(record.get("id")).thenReturn(1L);
+    Mockito.when(record.get("price")).thenReturn(20.2008);
+    Mockito.when(record.get("stockSize")).thenReturn(100);
+    Mockito.when(record.get("updated")).thenReturn(Boolean.TRUE);
+    Mockito.when(record.get("batchId")).thenReturn("12DER");
+    Mockito.when(record.get("tax")).thenReturn(50.0000);
+    Mockito.when(record.get("updatedPrice")).thenReturn(new BigDecimal(24.2).unscaledValue().toByteArray());
+    Mockito.when(record.get("timestampMillis")).thenReturn(1508484583259L);
+    Mockito.when(record.get("timestamp")).thenReturn(1508484583259L);
+    Mockito.when(record.get("timeMillis")).thenReturn(85398);
+    Mockito.when(record.get("timeMicros")).thenReturn(8638999L);
+    Mockito.when(record.get("dateTime")).thenReturn("2022-06-10T10:43:56");
+    Mockito.when(record.getSchema()).thenReturn(schema);
+    ServiceNowTransformer recordToJsonTransformer = new ServiceNowTransformer();
+    recordToJsonTransformer.transform(record);
```

Comment:
stored the JsonObject returned by transform() in a variable, then check the value of all the fields

Reply:
Done